aiida/backends/tests/engine/test_launch.py

```diff
@@ -273,3 +276,61 @@ def test_calcjob_dry_run_no_provenance(self):
         self.assertIn('folder', node.dry_run_info)
         for filename in ['single_file', 'file_one', 'file_two']:
             self.assertIn(filename, os.listdir(node.dry_run_info['folder']))
+
+
+class TestLauncherDisabledComputer(AiidaTestCase):
+    """Test the launcher when computer is disabled."""
+
+    @classmethod
+    def setUpClass(cls, *args, **kwargs):
+        """Set up a simple authinfo and for later use."""
+        super().setUpClass(*args, **kwargs)
+        cls.authinfo = orm.AuthInfo(computer=cls.computer, user=orm.User.objects.get_default())
+        cls.authinfo.store()
+
+    def setUp(self):
+        super().setUp()
+        self.assertIsNone(Process.current())
+        self.authinfo.enabled = False  # disable the computer
+
+    def tearDown(self):
+        super().tearDown()
+        self.assertIsNone(Process.current())
+        self.authinfo.enabled = True  # re-enable again
+
+    def test_calcjob_disabled_computer(self):
+        """Test that running a calcjob on a disabled computer will set the job to pause.
+
+        """
+        from aiida.plugins import CalculationFactory
+
+        ArithmeticAddCalculation = CalculationFactory('arithmetic.add')  # pylint: disable=invalid-name
+
+        code = orm.Code(
+            input_plugin_name='arithmetic.add',
+            remote_computer_exec=[self.computer, '/bin/true']).store()
+
+        inputs = {
+            'code': code,
+            'x': orm.Int(1),
+            'y': orm.Int(1),
+            'metadata': {
+                'options': {
+                    'resources': {'num_machines': 1, 'num_mpiprocs_per_machine': 1}
+                }
+            }
+        }
+
+        runner = get_manager().get_runner()
+
+        @gen.coroutine
+        def run_with_disabled():
+            calc_node = runner.submit(ArithmeticAddCalculation, **inputs)
+            for _ in range(1000):
```

Comment:
Is this loop just there to give the runner the chance to pause the process? In that case it might be better to use a small sleep, e.g.:
```
        timeout = 1  # Give the runner sometime to pause the calculation after submission
        runner = get_manager().get_runner()

        @gen.coroutine
        def run_with_disabled():
            calc_node = runner.submit(ArithmeticAddCalculation, **inputs)
            self.assertFalse(calc_node.paused)
            yield gen.sleep(timeout)
            self.assertTrue(calc_node.paused)

        runner.loop.run_sync(run_with_disabled)
```

Reply:
Yes, it is. No, I don't think that's an appropriate solution: while I can assume that we have a finite and sane amount of yields/process interrupts in-between I can't make any assumptions about how much time they take, a sufficiently slow worker might get back from the sleep first, no?

Reply:
That is my problem with this solution though, I have no idea how long a 1000 yields take. I prefer to have a sane explicit timeout in seconds so we now how long we are waiting before giving up. If you think 1 second is cutting it a bit short and runs the risks of giving up too early we can increase it.

Reply:
I am somewhat reluctant to build-in a race condition, but it's just a test, so ok (btw, would be max 1000 yields, in tests it takes less than 10). 1s should be more than enough, unless of course your system suddenly starts to swap, mines bitcoins or is encrypting your files ;-)